src/nxdoc/nuxeo-server/platform-services/comments.md

```diff
@@ -20,7 +20,73 @@ Comments (and so annotations) are linked to a specific document (their parent).
 
 ## Core Implementation
 
-### Storage
+### Current implementation - TreeCommentManager
+
+#### Storage
+
+Comments in Nuxeo are stored as regular documents of type `Comment`.</br>
+Comments hold:
+
+- parent id (legacy purpose)
+- ancestor ids (legacy purpose)
+- author
+- text
+- creation date
+- modification date
+
+[`Comment`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Comment.html) interface has been created to handle comments in Nuxeo. The interface is used as based type in [comment service](#java-api) and also defines how comments are serialized in JSON in order to use them with the [REST API](#json-entity-format).
+
+Annotations are stored as regular document of type `Annotation` which inherit from `Comment`. They hold in addition of comments:
+
+- blob xpath
+
+Like comments, annotations have the [`Annotation`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Annotation.html) interface to use them.
+
+Comments and annotations can have the facet `ExternalEntity`. When they have this facet, it can hold in addition:
+
+- external entity id
+- external entity
+- external entity origin</br>
+With this facet, it is possible to handle external content such as annotations made in [ARender]({{page space='nxdoc' page='nuxeo-arender-connector'}}) for instance.
+
+Comments and annotations are stored as a tree structure in a folderish document of type `CommentRoot` which is located right under the commented document. Comments are _special children_ which means they are not copied with the commented document during copy operations but they are snapshot when the document is checked in.</br>
+Please note comments are not restored when a version of the commented document is checked out.
+
+#### Permissions
+
+Comments and annotations rely on the the `ACL` security model present in Nuxeo. People who have access to a commented document have read access to its comments and can edit their own comments and annotations.
+
+#### Events
+
+All events share the same event context. It contains:
+
+- the parent document in `parentComment` which may be a comment or the commented document
+- the comment in `comment_document`
+- the comment text in `comment`
+- the category `commentCategory` in `category`
+
+A `commentAdded` event is fired when creating a comment or annotation.
+
```

Comment:
We have also the `commentUpdate`

Reply:
Perhaps we can remove the redundancy on `event is` as we are in the `Events` section

Reply:
About updated to be exact we end up calling: https://github.com/nuxeo/nuxeo/blob/2387c0ba417810c3b68fd8d51795b46cdee6f579/modules/platform/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/SubscriptionAdapter.java#L165

And it ends up firing every event registered here:
https://github.com/nuxeo/nuxeo/blob/2387c0ba417810c3b68fd8d51795b46cdee6f579/modules/platform/nuxeo-platform-comment/nuxeo-platform-comment/src/main/resources/OSGI-INF/comment-notification-contrib.xml#L11-L27

So we could even give a contrib example about how to contribute to those fired events. wdyt ?

"A `commentAdded` is fired" would feel weirder to me

Reply:
>"A commentAdded is fired" would feel weirder to me

Reply:
>"A commentAdded is fired" would feel weirder to me

Yep, looks good to me too 

>So we could even give a contrib example about how to contribute to those fired events. wdyt ?

Yep, why not. my main idea is give the information that update is also fired to avoid any interrogation from the reader